test/tools/cli/IonJavaCli.java

```diff
@@ -0,0 +1,1341 @@
+                    IonTimestamp ionTimestampSecond = (IonTimestamp) eventSecond.getValue();
+                    compareResult = ionTimestampFirst.timestampValue()
+                            .compareTo(ionTimestampSecond.timestampValue()) == 0;
+                } else {
+                    compareResult = Equivalence.ionEquals(eventFirst.getValue(), eventSecond.getValue());
+                }
+                if (!compareResult) {
+                    setReportInfo(i, j, eventFirst.getValue() + " vs. "
+                            + eventSecond.getValue(), compareContext);
+                    return false;
+                }
+            }
+            i++;
+            j++;
+        }
+        return true;
+    }
+
+    private static IonStruct parseStruct(ContainerContext containerContext,
+                                         CompareContext compareContext,
+                                         int end,
+                                         boolean isFirst) {
+        Event initEvent = isFirst ?
+                compareContext.getEventStreamFirst().get(containerContext.getIndex()) :
+                compareContext.getEventStreamSecond().get(containerContext.getIndex());
+        if (initEvent.getEventType() != EventType.CONTAINER_START || initEvent.getIonType() != IonType.STRUCT) {
+            return null;
+        }
+        IonStruct ionStruct = ION_SYSTEM.newEmptyStruct();
+        while (containerContext.increaseIndex() < end) {
+            Event event = isFirst ?
+                    compareContext.getEventStreamFirst().get(containerContext.getIndex()) :
+                    compareContext.getEventStreamSecond().get(containerContext.getIndex());
+            EventType eventType = event.getEventType();
+            if (eventType == EventType.CONTAINER_START) {
+                switch (event.getIonType()) {
+                    case LIST:
+                        ionStruct.add(event.getFieldName(),
+                                parseSequence(containerContext, compareContext, end, isFirst, ION_SYSTEM.newEmptyList()));
+                        break;
+                    case STRUCT:
+                        ionStruct.add(event.getFieldName(),
+                                parseStruct(containerContext, compareContext, end, isFirst));
+                        break;
+                    case SEXP:
+                        ionStruct.add(event.getFieldName(),
+                                parseSequence(containerContext, compareContext, end, isFirst, ION_SYSTEM.newEmptySexp()));
+                        break;
+                }
+            } else if (eventType == EventType.CONTAINER_END) {
+                break;
+            } else if (eventType == EventType.STREAM_END) {
+                throw new IonException("Invalid struct: eventStream ends without CONTAINER_END");
+            } else {
+                IonValue cloneValue = event.getValue().clone();
+                cloneValue.removeFromContainer();
```

Comment:
I believe cloning already removes the value from its container, so this line is probably redundant.